pallets/capacity/src/types.rs

```diff
@@ -101,6 +101,37 @@ impl<T: Config> StakingAccountDetails<T> {
 		total_reaped
 	}
 
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	#[allow(clippy::unwrap_used)]
+	/// for testing and benchmarks only!
+	/// set stake_change_unlocking chunks with (balance, thaw_at).  does not check that the unlock chunks
+	/// don't exceed total.
+	/// returns true on success, false on failure (?)
+	pub fn set_stake_change_unlock_chunks(&mut self, chunks: &Vec<(u32, u32)>) -> bool {
+		let result: Vec<UnlockChunk<BalanceOf<T>, <T>::EpochNumber>> = chunks
+			.into_iter()
+			.map(|chunk| UnlockChunk { value: chunk.0.into(), thaw_at: chunk.1.into() })
+			.collect();
+		self.unlocking = BoundedVec::try_from(result).unwrap();
+		self.unlocking.len() == chunks.len()
+	}
+
+	/// update unlock chunks; remove those that have expired and add the new one
+	// this doesn't affect staking amount, just controls how often a staker may retarget
```

Comment:
Just curious. Why do we want to limit the number of times a staker may retarget?

Reply:
I felt like it was a way to stabilize the capacity market, for lack of a better term; you should not be able to retarget endlessly - if a whale came in and had a bunch of stake to one or a handful of targets and then they redistributed a lot it could destabilize the ability to post messages very quickyl, not giving Provider(s) enough time to respond.  It is meant to discourage "insincere" retargeting.